src/semantics/Action.re

```diff
@@ -3408,6 +3420,47 @@ let rec syn_perform_block =
     let (ze, u_gen) = ZExp.new_EmptyHole(u_gen);
     let zblock = ZExp.BlockZE(lines, ze);
     syn_perform_block(~ci, ctx, keyword_action(k), (zblock, Hole, u_gen));
+
+  /* Variable Splitting */
+
+  | (Construct(SOp(op)), BlockZE(lines, zexp))
+      when zexp_is_suitable_for_var_split(zexp) =>
+    let split_var = (pos, err, verr, name) => {
+      let left_var = String.sub(name, 0, pos);
+      let right_var = String.sub(name, pos, String.length(name) - pos);
+      let cursor =
+        ZExp.CursorE(
+          OnText(String.length(name) - pos),
+          Var(err, verr, right_var),
+        );
+      let left_var = UHExp.Var(err, verr, left_var);
```

Comment:
You don't want to preserve the `err` and `verr` values here. `err: ErrStatus.t` represents the type-checked status of the original var (e.g. `InHole(TypeInconsistent, <hole-num>)` if the original var had type `Num` when there was a `Bool` expected). `verr: VarErrStatus.t` represents the variable-specific error status of the original var (e.g. `InVarHole(Free, <hole-num>)` if the var isn't bound, `InVarHole(Keyword, <hole-num>)` if it's a reserved keyword). These values no longer hold for the new split vars. What I'd do here instead is just use `UHExp.var` to construct the new split vars, then later you'll need to call `Statics.syn_fix_holes_zblock` on the final result zblock to fix the hole statuses.